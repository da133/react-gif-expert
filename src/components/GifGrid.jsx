import PropTypes from 'prop-types';
import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category, onDeleteCategory }) => {
    
    const { images, isLoading } = useFetchGifs( category );

    //Está función la agregue yo
    const deleteElement = () => {
        onDeleteCategory(category);
    };

    return (
        <>
            <h3>{ category }</h3>
            <button className="btn-delete" onClick={ deleteElement }>Delete</button>
            
            {
                isLoading && (<h2>Cargando...</h2>)
            }
            
            <div className="card-grid">
                {
                    images.map( ( image ) => (
                        <GifItem 
                            key={ image.id } 
                            { ...image }
                        />
                    ))
                }
            </div>        
        </>
    );
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
    onDeleteCategory: PropTypes.func
}