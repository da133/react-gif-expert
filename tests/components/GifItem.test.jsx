import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas de GifItem', () => {

    const title = 'The Office';
    const url = 'https://the-office/series.jpg';

    test('debe hacer match con el snapshot', () => {
        const { container } = render( <GifItem title={ title } url={ url } /> );
        expect( container ).toMatchSnapshot();
    });

    test('debe de mostrar la imagen con el URL y el ALT indicado', () => {
        render( <GifItem title={ title } url={ url } /> );
        // screen.debug();
        // expect( screen.getByRole('img').urÃ± ).toBe( url );
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
    });

    test('debe de mostrar el titulo en el componente', () => {
        render( <GifItem title={ title } url={ url } /> );
        console.log( screen.getByAltText( title ) );
        expect( screen.getByAltText( title ) ).toBeTruthy();
    });

});