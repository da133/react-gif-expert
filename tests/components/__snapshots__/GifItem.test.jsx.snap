// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifItem debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="The Office"
      src="https://the-office/series.jpg"
    />
  </div>
</div>
`;
